pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -145,11 +147,35 @@ func (r *ReconcileNMState) applyRBAC(instance *nmstatev1beta1.NMState) error {
 
 func (r *ReconcileNMState) applyHandler(instance *nmstatev1beta1.NMState) error {
 	data := render.MakeRenderData()
+	// Register toYaml function
+	data.Funcs["toYaml"] = helper.ToYaml
+
+	// Prepare defaults
+	masterExistsNoScheduleToleration := corev1.Toleration{
+		Key:      "node-role.kubernetes.io/master",
+		Operator: corev1.TolerationOpExists,
+		Value:    string(corev1.TaintEffectNoSchedule),
+	}
+	handlerNodeSelector := instance.Spec.NodeSelector
+	handlerNodeSelector["beta.kubernetes.io/arch"] = "amd64"
+
 	data.Data["HandlerNamespace"] = os.Getenv("HANDLER_NAMESPACE")
 	data.Data["HandlerImage"] = os.Getenv("HANDLER_IMAGE")
 	data.Data["HandlerPullPolicy"] = os.Getenv("HANDLER_IMAGE_PULL_POLICY")
 	data.Data["HandlerPrefix"] = os.Getenv("HANDLER_PREFIX")
-	data.Data["HandlerNodeSelector"] = instance.Spec.NodeSelector
+	data.Data["WebhookNodeSelector"] = map[string]string{
+		"beta.kubernetes.io/arch":        "amd64",
+		"node-role.kubernetes.io/master": "",
+	}
+	data.Data["WebhookTolerations"] = []corev1.Toleration{
+		masterExistsNoScheduleToleration,
+	}
+	data.Data["WebhookAffinity"] = corev1.Affinity{}
+	data.Data["HandlerNodeSelector"] = handlerNodeSelector
```

Comment:
Use directly `instance.Spec.NodeSelector` here.

Reply:
Just a question here, the handler nodeselector originally contained `beta.kubernetes.io/arch: amd64` + the content of `instance.Spec.NodeSelector`. In this commit, it is `only instance.Spec.NodeSelector`, is it okay?

Reply:
Nope, we have to add amd64 to the stuff that comes from `instance.Spec.NodeSelector` I suppose it's spected that amd64 is a safe default for the NMstate CR, since we have containers only for amd64.

Reply:
Okay, that was the point of using the intermediate handlerNodeSelector variable.
Changed back and hopefully named the var better